### DISCUSSION TITLE
How to use Directory.glob?

### DISCUSSION THREAD

**Discussion by skycaptain** [dagger/dagger#7156] [Help] [answered]
I may be missing something simple, but I'm currently struggling with `Directory.glob` returning an empty list instead of the files. Consider this example where I want to match all the files in the `deploy/sdk` directory. Note the different `tmp-*` directories:

```python
mydir = (
    dag.directory()
    .with_new_file("build/tmp-build-a/deploy/sdk/test.txt", "FOO")
    .with_new_file("build/tmp-build-a/deploy/sdk/test.manifest", "BAR")
    .with_new_file("build/tmp-build-b/deploy/ipk/test.ipk", "BAZ")
)

await mydir.glob('build/tmp-*/deploy/sdk/*')
```

This example returns `[]`. Running the same command with regular `ls` works fine. I tried different combinations of the pattern, but none have worked as expected. What could I be missing?

**Comment by marcosnils** [comment, marked as answer]
:wave:  seems like a bug in the `glob` pattern matching function. According to this: https://github.com/moby/patternmatcher/blob/main/patternmatcher_test.go#L140 that pattern should work. I even cloned that repository and added the following test case: 

```go 
	tests := []matchesTestCase{
		{"build/tmp-*/deploy/sdk*", "build/tmp-build-a/deploy/sdk/test.manifest", true},
```
and this passed. 

It's very likely an issue coming from https://github.com/marcosnils/dagger/blob/9baaa467f14bd4abf5399da1608b9d17d07e8fd7/core/directory.go?plain=1#L272

just opened and issue to track this: https://github.com/dagger/dagger/issues/7169

cc @TomChv

**Reply by skycaptain** [reply]
Since #7169 is done and glob now works as expected we can close this dicussion
